app/models/user.rb

```diff
@@ -129,6 +129,7 @@ class User < ApplicationRecord
   before_validation :sanitize_role
   before_create :set_approved
   after_commit :send_pending_devise_notifications
+  after_find :set_role_id
```

Comment:
Would it make sense to migrate existing records? I'm afraid if `after_find` is going to cover all the cases.